src/compas_fd/fd/fd_iter_numpy.py

```diff
@@ -0,0 +1,101 @@
+from typing import Any
+from typing import Tuple
+from typing import List
+from typing import Union
+from typing import Sequence
+from typing import Optional
+from typing_extensions import Annotated
+from nptyping import NDArray
+
+from dataclasses import astuple
+
+from numpy import asarray
+from numpy import float64
+from scipy.linalg import norm
+from scipy.sparse.linalg import spsolve
+
+from compas.numerical import normrow
+
+from .fd_numerical_data import FDNumericalData
+from .result import Result
+from ..constraints import Constraint
+
+
+def fd_iter_numpy(*,
+                  vertices: Union[Sequence[Annotated[List[float], 3]], NDArray[(Any, 3), float64]],
+                  fixed: List[int],
+                  edges: List[Tuple[int, int]],
+                  forcedensities: List[float],
+                  loads: Optional[Union[Sequence[Annotated[List[float], 3]], NDArray[(Any, 3), float64]]] = None,
+                  constraints: Sequence[Constraint],
+                  max_iter: int = 100, tol_res: float = 1E-3, tol_xyz: float = 1E-3
+                  ) -> Result:
+    """Iteratively compute the equilibrium coordinates of a system of vertices connected by edges.
+    Vertex constraints are recomputed at each iteration.
+    """
+    nd = FDNumericalData.from_fd_params(vertices, fixed, edges, forcedensities, loads)
+
+    for k in range(max_iter):
+        _xyz = nd.xyz
```

Comment:
why the underscore?

Reply:
to differentiate it as the previous iteration xyz,
I will rename it to xyz_prev